File: Abstraction/Java/BasicCalculator.java
package Abstraction.Java;

// BasicCalculator class that extends the abstract class Calculator
public class BasicCalculator extends Calculator {
    // Override method to add a value to the result
    @Override
    public void add(int value) {
        result += value;
    }

    // Override method to subtract a value from the result
    @Override
    public void subtract(int value) {
        result -= value;
    }

    // Main method to test the BasicCalculator class
    public static void main(String[] args) {
        // Create an instance of BasicCalculator
        BasicCalculator calc = new BasicCalculator();

        // Test add method
        calc.add(10);
        System.out.println("Result after adding 10: " + calc.getResult());

        // Test subtract method
        calc.subtract(5);
        System.out.println("Result after subtracting 5: " + calc.getResult());

        // Test reset method
        calc.reset();
        System.out.println("Result after reset: " + calc.getResult());
    }
}

File: Abstraction/Javascript/basicCalculator.js
// Import the Calculator class
const Calculator = require('./calculator');

// BasicCalculator class that extends the abstract class Calculator
class BasicCalculator extends Calculator {
    // Method to add a value to the result
    add(value) {
        this._result += value;
    }

    // Method to subtract a value from the result
    subtract(value) {
        this._result -= value;
    }
}

// Export the BasicCalculator class
module.exports = BasicCalculator;

// Test the BasicCalculator class
if (require.main === module) {
    const calc = new BasicCalculator();
    
    calc.add(10);
    console.log(`Result after adding 10: ${calc.getResult()}`); // Output: Result after adding 10: 10
    
    calc.subtract(5);
    console.log(`Result after subtracting 5: ${calc.getResult()}`); // Output: Result after subtracting 5: 5
    
    calc.reset();
    console.log(`Result after reset: ${calc.getResult()}`); // Output: Result after reset: 0
}
